lib/template/details-json.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+	function getDetails($aLine){
+		$aDetails = array();
+		if ($aLine['place_id'])
+		{
+			$aDetails['place_id'] = $aLine['place_id'];
+		}
+		$sOSMType = ($aLine['osm_type'] == 'N'?'node':($aLine['osm_type'] == 'W'?'way':($aLine['osm_type'] == 'R'?'relation':($aLine['osm_type'] == 'T'?'tiger':''))));
+		if ($sOSMType)
+		{
+			$aDetails['osm_type'] = $sOSMType;
+			$aDetails['osm_id'] = $aLine['osm_id'];
+		}
+		if ($aLine['localname']) {
+			$aDetails['name'] = $aLine['localname'];
+		}
+		$aDetails['class'] = $aLine['class'];
+		$aDetails['type'] = $aLine['type'];
+		if ($aLine['housenumber']) {
+			$aDetails['housenumber'] = $aLine['housenumber'];
+		}
+		return $aDetails;
+	}
+
+	function wikipedia_link($sWikipedia)
+	{
+		if ($sWikipedia)
+		{
+			list($sWikipediaLanguage,$sWikipediaArticle) = explode(':',$sWikipedia);
+			return 'https://'.$sWikipediaLanguage.'.wikipedia.org/wiki/'.urlencode($sWikipediaArticle);
+		}
+		return '';
+	}
```

Comment:
code duplication from details-html.php. It's time to start a new file lib/output.php and move all the functions from details-*.php in there.